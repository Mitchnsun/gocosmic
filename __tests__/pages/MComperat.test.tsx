import MComperat from '@/app/[locale]/projects/mcomperat/page';

import { render } from '../test-utils';

describe('MComperat Page', () => {
  it('should render the mcomperat page correctly', () => {
    const { container } = render(<MComperat />);

    expect(container).toMatchSnapshot();
  });
});
