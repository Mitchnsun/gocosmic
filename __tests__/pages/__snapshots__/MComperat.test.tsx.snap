// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MComperat Page > should render the mcomperat page correctly 1`] = `
<div>
  <div
    class="text-ghost relative pt-10"
  >
    <main
      class="m-auto flex max-w-7xl flex-col items-center gap-10 px-4 pb-4"
    >
      <div
        class="flex flex-col items-center justify-between gap-8"
      >
        <div
          class="text-center"
        >
          <h1
            class="mb-4 text-2xl font-extrabold sm:text-4xl"
          >
            mcomper.at
          </h1>
          <p
            class="text-lg text-gray-400"
          >
            Modern, responsive web CV of Matthieu Compérat
          </p>
        </div>
      </div>
      <section
        aria-labelledby="overview-heading"
        class="w-full"
      >
        <div
          class="relative rounded-lg bg-gray-800 px-6 py-8 lg:p-8"
        >
          <div
            class="mb-6 flex items-center gap-4"
          >
            <svg
              aria-hidden="true"
              class="h-8 w-8 shrink-0 text-purple-400"
              data-slot="icon"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.721 12.752a9.711 9.711 0 0 0-.945-5.003 12.754 12.754 0 0 1-4.339 2.708 18.991 18.991 0 0 1-.214 4.772 17.165 17.165 0 0 0 5.498-2.477ZM14.634 15.55a17.324 17.324 0 0 0 .332-4.647c-.952.227-1.945.347-2.966.347-1.021 0-2.014-.12-2.966-.347a17.515 17.515 0 0 0 .332 4.647 17.385 17.385 0 0 0 5.268 0ZM9.772 17.119a18.963 18.963 0 0 0 4.456 0A17.182 17.182 0 0 1 12 21.724a17.18 17.18 0 0 1-2.228-4.605ZM7.777 15.23a18.87 18.87 0 0 1-.214-4.774 12.753 12.753 0 0 1-4.34-2.708 9.711 9.711 0 0 0-.944 5.004 17.165 17.165 0 0 0 5.498 2.477ZM21.356 14.752a9.765 9.765 0 0 1-7.478 6.817 18.64 18.64 0 0 0 1.988-4.718 18.627 18.627 0 0 0 5.49-2.098ZM2.644 14.752c1.682.971 3.53 1.688 5.49 2.099a18.64 18.64 0 0 0 1.988 4.718 9.765 9.765 0 0 1-7.478-6.816ZM13.878 2.43a9.755 9.755 0 0 1 6.116 3.986 11.267 11.267 0 0 1-3.746 2.504 18.63 18.63 0 0 0-2.37-6.49ZM12 2.276a17.152 17.152 0 0 1 2.805 7.121c-.897.23-1.837.353-2.805.353-.968 0-1.908-.122-2.805-.353A17.151 17.151 0 0 1 12 2.276ZM10.122 2.43a18.629 18.629 0 0 0-2.37 6.49 11.266 11.266 0 0 1-3.746-2.504 9.754 9.754 0 0 1 6.116-3.985Z"
              />
            </svg>
            <h2
              class="text-xl font-bold sm:text-2xl lg:text-3xl"
              id="overview-heading"
            >
              About the Project
            </h2>
          </div>
          <div
            class="space-y-4 text-gray-300"
          >
            <p
              class="text-lg"
            >
              Modern and responsive web CV of Matthieu Compérat, designed with Next.js (App Router) for fast, accessible, and SEO-friendly pages.
            </p>
            <p>
              Clear and maintainable architecture: reusable components, Server Components by default, and strict types for production reliability. Integrated internationalization in French (default) and English, with homepage redirection to /fr.
            </p>
          </div>
        </div>
      </section>
      <section
        aria-labelledby="features-heading"
        class="w-full"
      >
        <div
          class="rounded-lg bg-gray-800 px-6 py-8 lg:p-8"
        >
          <div
            class="mb-6 flex items-center gap-4"
          >
            <svg
              aria-hidden="true"
              class="h-8 w-8 shrink-0 text-yellow-400"
              data-slot="icon"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M9 4.5a.75.75 0 0 1 .721.544l.813 2.846a3.75 3.75 0 0 0 2.576 2.576l2.846.813a.75.75 0 0 1 0 1.442l-2.846.813a3.75 3.75 0 0 0-2.576 2.576l-.813 2.846a.75.75 0 0 1-1.442 0l-.813-2.846a3.75 3.75 0 0 0-2.576-2.576l-2.846-.813a.75.75 0 0 1 0-1.442l2.846-.813A3.75 3.75 0 0 0 7.466 7.89l.813-2.846A.75.75 0 0 1 9 4.5ZM18 1.5a.75.75 0 0 1 .728.568l.258 1.036c.236.94.97 1.674 1.91 1.91l1.036.258a.75.75 0 0 1 0 1.456l-1.036.258c-.94.236-1.674.97-1.91 1.91l-.258 1.036a.75.75 0 0 1-1.456 0l-.258-1.036a2.625 2.625 0 0 0-1.91-1.91l-1.036-.258a.75.75 0 0 1 0-1.456l1.036-.258a2.625 2.625 0 0 0 1.91-1.91l.258-1.036A.75.75 0 0 1 18 1.5ZM16.5 15a.75.75 0 0 1 .712.513l.394 1.183c.15.447.5.799.948.948l1.183.395a.75.75 0 0 1 0 1.422l-1.183.395c-.447.15-.799.5-.948.948l-.395 1.183a.75.75 0 0 1-1.422 0l-.395-1.183a1.5 1.5 0 0 0-.948-.948l-1.183-.395a.75.75 0 0 1 0-1.422l1.183-.395c.447-.15.799-.5.948-.948l.395-1.183A.75.75 0 0 1 16.5 15Z"
                fill-rule="evenodd"
              />
            </svg>
            <h2
              class="text-xl font-bold sm:text-2xl lg:text-3xl"
              id="features-heading"
            >
              Key Features
            </h2>
          </div>
          <div
            class="space-y-4 text-gray-300"
          >
            <ul
              class="grid gap-3 md:grid-cols-2"
            >
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-yellow-400"
                />
                <span>
                  Internationalization: /fr and /en routes with dedicated JSON content
                </span>
              </li>
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-yellow-500"
                />
                <span>
                  SEO: metadata managed via Next.js API (generateMetadata) for better ranking
                </span>
              </li>
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-yellow-600"
                />
                <span>
                  Performance: Server Components, streaming, and React 19 for fast loading
                </span>
              </li>
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-yellow-700"
                />
                <span>
                  Accessibility: best practices (jsx-a11y) and semantic HTML
                </span>
              </li>
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-yellow-800"
                />
                <span>
                  Code quality: ESLint (flat config), Prettier (Tailwind sort), Husky hooks, sorted imports, zero lint errors
                </span>
              </li>
            </ul>
          </div>
        </div>
      </section>
      <section
        aria-labelledby="technology-heading"
        class="w-full"
      >
        <div
          class="rounded-lg bg-gray-800 px-6 py-8 lg:p-8"
        >
          <div
            class="mb-6 flex items-center gap-4"
          >
            <svg
              aria-hidden="true"
              class="h-8 w-8 shrink-0 text-blue-400"
              data-slot="icon"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M14.447 3.026a.75.75 0 0 1 .527.921l-4.5 16.5a.75.75 0 0 1-1.448-.394l4.5-16.5a.75.75 0 0 1 .921-.527ZM16.72 6.22a.75.75 0 0 1 1.06 0l5.25 5.25a.75.75 0 0 1 0 1.06l-5.25 5.25a.75.75 0 1 1-1.06-1.06L21.44 12l-4.72-4.72a.75.75 0 0 1 0-1.06Zm-9.44 0a.75.75 0 0 1 0 1.06L2.56 12l4.72 4.72a.75.75 0 0 1-1.06 1.06L.97 12.53a.75.75 0 0 1 0-1.06l5.25-5.25a.75.75 0 0 1 1.06 0Z"
                fill-rule="evenodd"
              />
            </svg>
            <h2
              class="text-xl font-bold sm:text-2xl lg:text-3xl"
              id="technology-heading"
            >
              Technology Stack
            </h2>
          </div>
          <div
            class="space-y-4 text-gray-300"
          >
            <p
              class="text-lg"
            >
              Built with cutting-edge web technologies for a fast, maintainable, and production-ready experience:
            </p>
            <ul
              class="grid gap-3 md:grid-cols-2"
            >
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-blue-400"
                />
                <span>
                  Next.js 15.5+ (App Router) - Modern React framework
                </span>
              </li>
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-blue-500"
                />
                <span>
                  TypeScript 5.9+ (strict mode) - Type safety and reliability
                </span>
              </li>
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-blue-600"
                />
                <span>
                  Tailwind CSS 4.1+ - Utility-first styling with custom theme
                </span>
              </li>
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-blue-700"
                />
                <span>
                  Yarn 4+ (Corepack) and Node.js 22+ - Modern tooling
                </span>
              </li>
            </ul>
          </div>
        </div>
      </section>
      <section
        aria-labelledby="ai-heading"
        class="w-full"
      >
        <div
          class="rounded-lg bg-gray-800 px-6 py-8 lg:p-8"
        >
          <div
            class="mb-6 flex items-center gap-4"
          >
            <svg
              aria-hidden="true"
              class="h-8 w-8 shrink-0 text-green-400"
              data-slot="icon"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.5 7.5h-9v9h9v-9Z"
              />
              <path
                clip-rule="evenodd"
                d="M8.25 2.25A.75.75 0 0 1 9 3v.75h2.25V3a.75.75 0 0 1 1.5 0v.75H15V3a.75.75 0 0 1 1.5 0v.75h.75a3 3 0 0 1 3 3v.75H21A.75.75 0 0 1 21 9h-.75v2.25H21a.75.75 0 0 1 0 1.5h-.75V15H21a.75.75 0 0 1 0 1.5h-.75v.75a3 3 0 0 1-3 3h-.75V21a.75.75 0 0 1-1.5 0v-.75h-2.25V21a.75.75 0 0 1-1.5 0v-.75H9V21a.75.75 0 0 1-1.5 0v-.75h-.75a3 3 0 0 1-3-3v-.75H3A.75.75 0 0 1 3 15h.75v-2.25H3a.75.75 0 0 1 0-1.5h.75V9H3a.75.75 0 0 1 0-1.5h.75v-.75a3 3 0 0 1 3-3h.75V3a.75.75 0 0 1 .75-.75ZM6 6.75A.75.75 0 0 1 6.75 6h10.5a.75.75 0 0 1 .75.75v10.5a.75.75 0 0 1-.75.75H6.75a.75.75 0 0 1-.75-.75V6.75Z"
                fill-rule="evenodd"
              />
            </svg>
            <h2
              class="text-xl font-bold sm:text-2xl lg:text-3xl"
              id="ai-heading"
            >
              Powered by AI
            </h2>
          </div>
          <div
            class="space-y-4 text-gray-300"
          >
            <p
              class="text-lg"
            >
              Development workflow enhanced by AI assistance for faster, more consistent, and higher quality results:
            </p>
            <ul
              class="grid gap-3 md:grid-cols-2"
            >
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-green-400"
                />
                <span>
                  Assisted development: recommended use of GitHub Copilot to accelerate component writing and reduce errors
                </span>
              </li>
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-green-500"
                />
                <span>
                  Quality and consistency: automatic suggestions for import fixes, ESLint/TypeScript compliance, and Tailwind class ordering
                </span>
              </li>
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-green-600"
                />
                <span>
                  Content and SEO: assistance with microcopy, meta/OG tags, and accessible alt text
                </span>
              </li>
              <li
                class="flex items-center gap-3"
              >
                <span
                  aria-hidden="true"
                  class="mt-1 h-2 w-2 flex-shrink-0 rounded-full bg-green-700"
                />
                <span>
                  Internationalization: help pre-generating consistent FR/EN strings from templates, with mandatory human review
                </span>
              </li>
            </ul>
          </div>
        </div>
      </section>
      <section
        aria-labelledby="cta-heading"
        class="flex w-full flex-col items-center justify-center gap-8 lg:flex-row"
      >
        <div
          class="flex max-w-5xl flex-col items-center gap-6 rounded-lg bg-gray-800 px-6 py-12 lg:p-12"
        >
          <h2
            class="text-center text-3xl font-bold lg:text-4xl"
            id="cta-heading"
          >
            Explore mcomper.at
          </h2>
          <p
            class="text-center text-lg text-gray-400"
          >
            Discover Matthieu Compérat's professional web CV and explore his expertise.
          </p>
          <a
            aria-label="Visit mcomper.at website (opens in a new tab)"
            class="bg-royal hover:bg-royal/90 inline-flex items-center gap-2 rounded-full px-6 py-2 text-base font-normal text-neutral-50 transition-colors"
            href="https://www.mcomper.at/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Visit the website
            <svg
              aria-hidden="true"
              class="h-5 w-5"
              data-slot="icon"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M15.75 2.25H21a.75.75 0 0 1 .75.75v5.25a.75.75 0 0 1-1.5 0V4.81L8.03 17.03a.75.75 0 0 1-1.06-1.06L19.19 3.75h-3.44a.75.75 0 0 1 0-1.5Zm-10.5 4.5a1.5 1.5 0 0 0-1.5 1.5v10.5a1.5 1.5 0 0 0 1.5 1.5h10.5a1.5 1.5 0 0 0 1.5-1.5V10.5a.75.75 0 0 1 1.5 0v8.25a3 3 0 0 1-3 3H5.25a3 3 0 0 1-3-3V8.25a3 3 0 0 1 3-3h8.25a.75.75 0 0 1 0 1.5H5.25Z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </div>
      </section>
    </main>
  </div>
</div>
`;
